That should work, but isn't it a bit much to suppress the whole warning for 500+ lines of code, which could hide actual bugs in the future?

I think explicitly initializing the pointer here should be fine, with something like:
```diff
diff --git a/scene/resources/packed_scene.cpp b/scene/resources/packed_scene.cpp
index d3d12fb329..5ceb4c56fb 100644
--- a/scene/resources/packed_scene.cpp
+++ b/scene/resources/packed_scene.cpp
@@ -181,6 +181,7 @@ Node *SceneState::instantiate(GenEditState p_edit_state) const {
 	const NodeData *nd = &nodes[0];
 
 	Node **ret_nodes = (Node **)alloca(sizeof(Node *) * nc);
+	ret_nodes[0] = nullptr;
 
 	bool gen_node_path_cache = p_edit_state != GEN_EDIT_STATE_DISABLED && node_path_cache.is_empty();

```

---

The warning for context, from GCC 15.2.0 on Windows:
```
ERROR: scene\resources\packed_scene.cpp: In member function 'Node* SceneState::instantiate(GenEditState) const':
scene\resources\packed_scene.cpp:702:27: error: '*ret_nodes' may be used uninitialized [-Werror=maybe-uninitialized]
  702 |         return ret_nodes[0];
      |                           ^
cc1plus.exe: all warnings being treated as errors
```

It does seem a priori like a false positive.

--- a/core/math/geometry_2d.cpp
+++ b/core/math/geometry_2d.cpp
@@ -30,7 +30,9 @@
 
 #include "geometry_2d.h"
 
+GODOT_GCC_WARNING_PUSH_AND_IGNORE("-Walloc-zero")
 #include "thirdparty/clipper2/include/clipper2/clipper.h"
+GODOT_GCC_WARNING_POP
 #include "thirdparty/misc/polypartition.h"
 #define STB_RECT_PACK_IMPLEMENTATION
 #include "thirdparty/misc/stb_rect_pack.h"

--- a/scene/resources/packed_scene.cpp
+++ b/scene/resources/packed_scene.cpp
@@ -181,6 +181,7 @@ Node *SceneState::instantiate(GenEditState p_edit_state) const {
 	const NodeData *nd = &nodes[0];
 
 	Node **ret_nodes = (Node **)alloca(sizeof(Node *) * nc);
+	ret_nodes[0] = nullptr; // Sidesteps "maybe uninitialized" false-positives on GCC.
 
 	bool gen_node_path_cache = p_edit_state != GEN_EDIT_STATE_DISABLED && node_path_cache.is_empty();
 

